core-java-modules/core-java-uuid/src/test/java/com/baeldung/uuid/UUIDPositiveLongGeneratorUnitTest.java

```diff
@@ -2,42 +2,22 @@
 
 import org.junit.jupiter.api.Test;
 
-import java.lang.reflect.Method;
-import java.util.HashSet;
-import java.util.Set;
+import java.util.UUID;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class UUIDPositiveLongGeneratorUnitTest {
 
-    private final UUIDPositiveLongGenerator uuidLongGenerator = new UUIDPositiveLongGenerator();
-
-    private final Set<Long> uniqueValues = new HashSet<>();
-
     @Test
-    void whenForeachMethods_thenRetryWhileNotUnique() throws Exception {
-        for (Method method : uuidLongGenerator.getClass().getDeclaredMethods()) {
-            long uniqueValue;
-            do uniqueValue = (long) method.invoke(uuidLongGenerator); while (!isUnique(uniqueValue));
-            assertThat(uniqueValue).isPositive();
-        }
+    public void whengetMostSignificantBits_thenAssertPositive() {
+        long randomPositiveLong = Math.abs(UUID.randomUUID().getMostSignificantBits());
+        assertThat(randomPositiveLong).isPositive();
     }
 
     @Test
-    void whenGivenLongValue_thenCheckUniqueness() {
-        long uniqueValue = generateUniqueLong();
-        assertThat(uniqueValue).isPositive();
-    }
-
-    private long generateUniqueLong() {
-        long uniqueValue;
-        do uniqueValue = uuidLongGenerator.combineBitwise(); while (!isUnique(uniqueValue));
-        return uniqueValue;
-    }
-
-    private boolean isUnique(long value) {
-        // Implement uniqueness checking logic, for example, by checking in the database
-        return uniqueValues.add(value);
+    public void whengetLeastSignificantBits_thenAssertPositive() {
+        long randomPositiveLong = Math.abs(UUID.randomUUID().getLeastSignificantBits());
+        assertThat(randomPositiveLong).isPositive();
```

Comment:
As in https://github.com/eugenp/tutorials/pull/15692/files let's use isNotNegative, since this can technically be zero too